gnovm/pkg/gnolang/preprocess.go

```diff
@@ -3039,6 +3050,18 @@ func checkIntegerType(xt Type) {
 	}
 }
 
+func checkAssignmentMismatch(d Decl) {
+	if cd, ok := d.(*ValueDecl); ok {
```

Comment:
1. can we assert this before invoke this function?
2. I prefer using a name like assertXXX instead of checkXXX, what do you think?

Reply:
> can we assert this before invoke this function?

What do you mean here ?

> I prefer using a name like assertXXX instead of checkXXX, what do you think?

I also prefer to use assertXXX, but I noticed that in this file we are using checkXXX instead of assertXXX. I have decided to stick with checkXXX for consistency.

Reply:
I mean something like this, since we only want to check against `ValueDecl`,
```go
if cd, ok := d.(*ValueDecl); ok {
    checkAssignmentMismatch(d *ValueDecl)
    ...
}
```

or even 
```go
case *ValueDecl:
    // do check staff
```

but I think the first one is just ok.

Reply:
done here: https://github.com/gnolang/gno/pull/2206/commits/e45a6efc65d31a0ef5adb5b30c3e093147bdddad